### PR TITLE
fix of copy, cut and paste functionality.

### PR DESCRIPTION
**Pull request by kia**
code for copy, cut and paste added.

### CODE REVIEW

File: src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import javafx.scene.input.KeyCodeCombination;
+import javafx.scene.input.KeyCombination;
+import javafx.scene.input.KeyEvent;
+
+/**
+ * Wrapper class to make the clipboard functionality testable.
+ */
+public class SystemClipboardWrapper {
+
+	private final KeyCodeCombination KEY_CODE_CTRL_C = new KeyCodeCombination(KeyCode.C, KeyCombination.CONTROL_DOWN);
```
Comment (MegaCoderX) [inline src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java:16]: Is KeyCombination.CONTROL_DOWN the correct choice here? I am not 100% sure, but I believe in order to work on a MAC keyboard what should be used is [KeyCombination.SHORTCUT_DOWN](https://docs.oracle.com/javase/10/docs/api/javafx/scene/input/KeyCombination.html#SHORTCUT_DOWN)

Comment (miho) [reply src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java:16]: I am not sure. I can try on a Mac though. But I see nothing that prevents us from merging. Thanks @kia for doing the work!

Comment (kia) [reply src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java:16]: You are welcome @miho !

**Comment by treimers** [comment]
Hi all,

sorry, just discovered your discussion here. A little bit too late.

I think the portable approach for Copy (C), Cut (X) and Paste (V) is SHORTCUT instead of CONTROL.
This will be translated to ⌘-C, ⌘-X and ⌘-V on Mac OS X and Ctrl-C , Ctrl-X and Ctrl-V on Windows and Linux.

Have a look to this [description in the Java FX API](https://docs.oracle.com/javase/8/javafx/api/index.html?javafx/scene/input/KeyCombination.html):


The shortcut modifier is used to represent the modifier key which is used commonly in keyboard shortcuts on the host platform. This is for example control on Windows and meta (command key) on Mac. By using shortcut key modifier developers can create platform independent shortcuts. So the "Shortcut+C" key combination is handled internally as "Ctrl+C" on Windows and "Meta+C" on Mac.

File: src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java (hunk 1 above)
Comment (kia) [reply src/main/java/eu/mihosoft/monacofx/SystemClipboardWrapper.java:16]: @MegaCoderX thanks for the hint. There is a pull request now with the change you suggested.

**Comment by SunSiwen** [comment]
Hi All,

There is still one bug about `paste`.

It happens when you try to append on a new line:

image

As the picture shows:  and I tried to paste by the context-menu

image

And nothing happens. It is easy to reproduce the bug. Please do not hesitate to contact me if you have any question about my description.

Best regards,

Siwen Sun
Gmail: ssw1313113@gmail.com

**Comment by SunSiwen** [comment]
Hi  @miho @MegaCoderX  ,

I also found another buy about copy:

image

Select all, and then click the outside area to make it unactive.

image

Move mouse back to the text and righc click and copy

image


You can open any text file and try to paste. Nothing in your clipboard. The same opeartions applied to cut, the content is still here after cutting

I think it is very easy to fix. Please do not hesitate to contact me if you have any question about my description.

Best regards,

Siwen Sun
Gmail: [ssw1313113@gmail.com](mailto:ssw1313113@gmail.com)

**Pull request merged**